Refactor/naming (#550)
feature-sliced/documentation: i18n/en/docusaurus-plugin-content-docs/current/about/understanding/naming.md

# Naming

Different developers have different experiences and contexts, which can lead to misunderstandings on the team when the same entities are called differently. For example:

- Components for display can be called "ui", "components", "ui-kit", "views", …
- The code that is reused throughout the application can be called "core", "shared", "app", …
- Business logic code can be called "store", "model", "state", …

## Naming in Feature-Sliced Design {#naming-in-fsd}

The methodology uses specific terms such as:

- "app", "process", "page", "feature", "entity", "shared" as layer names,
- "ui', "model", "lib", "api", "config" as segment names.

> Original naming is crucial to prevent confusion among team members and new developers joining the project. Using unique names also helps when seeking help from the community.

Reviewer: issue: I think the meaning has slightly changed here from the previous version, and I think that the previous version was more on point here:

* Before: when developing a project, it is important to adhere to the original naming
* After: original naming is important

The difference here is subtle, to me, the Before version conveys "don't change the naming" and the After version conveys "use naming that hasn't been used before".

suggestion:
```suggestion
It is crucial to preserve these terms to prevent confusion among team members and new developers joining the project. Using standard names also helps when seeking help from the community.
```

Author: Fixed.

## Naming Conflicts {#when-can-naming-interfere}

Naming conflicts can occur when terms used in the FSD methodology overlap with terms used in the business:

- `FSD#process` vs simulated process in an application,
- `FSD#page` vs log page,
- `FSD#model` vs car model.

> For example, a developer seeing the word process in the code will spend unnecessary time trying to understand which process they are talking about.

Reviewer: suggestion (non-blocking): for clarity, to distinguish between process, _the word_, and a process _of words_
```suggestion
For example, a developer seeing the word "process" in the code will spend unnecessary time trying to understand which process they are talking about.
```

Reviewer: Also, I think that in this case, the developer in question is not talking about anything, they are reading someone else talking) so perhaps we could phrase it like this instead:
```suggestion
For example, a developer seeing the word "process" in the code will spend unnecessary time trying to understand which process is implied.
```

> These **collisions can disrupt the development process**, as developers may spend extra time trying to figure out exactly what they are talking about.

Reviewer: issue: same thing here about talking vs reading. However, in this case, this paragraph is simply restating what was said before, so perhaps we could join them?

```
For example, a developer seeing the word "process" in the code will spend unnecessary time trying to understand which process is implied. Such **collisions can disrupt development**.
```

Author: Fixed in latest commit.

When the project glossary contains terminology specific to FSD, it is critical to exercise caution when discussing these terms with the team and non-technical stakeholders.

To communicate effectively with the team, it is recommended that the abbreviation "FSD" be used to prefix the methodology terms. For example, when talking about a process, you might say, "We can put this process at the FSD level."

> Conversely, when communicating with non-technical stakeholders, it is best to limit the use of FSD terminology and refrain from mentioning the internal structure.

Reviewer: issue (non-blocking): I think it's a bit unclear here what internal structure we're talking about

suggestion:
```suggestion
Conversely, when communicating with non-technical stakeholders, it is best to limit the use of FSD terminology and refrain from mentioning the internal structure of the codebase.
```

Author: Fixed in latest commit.

# Naming

> Different developers have different experiences and contexts, which can lead to misunderstandings on the team when the same entities [are called differently][disc-src]. For example:

Reviewer: issue: in the original version, there was an abundance of links, I find them quite distracting (and also duplicated)

suggestion: remove the links from the text altogether, they're not adding a lot of meaning here
```suggestion
Different developers have different experiences and contexts, which can lead to misunderstandings on the team when the same entities are called differently. For example:
```

Author: Sure.

- Components for display can be called "ui", "components", "ui-kit", "views", ...
- The code that is reused throughout the application can be called "core", "shared", "app", ...
- Business logic code can be called "store", "model", "state", ...

## Naming in Feature-Sliced Design {#naming-in-fsd}

The methodology uses specific terms such as:

- [Layers][refs-layers] — "app", "process", "page", "feature", "entity", "shared"

> - [Slices][refs-slices]

Reviewer: issue: I don't think this item belongs here. The original intention was to highlight that there were some "reserved" keywords among layers and segments, the slices don't have them.

suggestion: 
```markdown
- "app", "process", "page", "feature", "entity", "shared" as layer names
- "ui', "model", "lib", "api", "config" as segment names
```

Author: Yes, that looks good

- [Segments][refs-segments] — "ui', "model", "lib", "api", "config"

> These terms are [clearly defined][refs-reference] in the methodology.

Reviewer: suggestion: let's just remove this line :)

> When working on a project based on the **Feature-Sliced Design** methodology, it is important to use [original naming][disc-src] to avoid confusion among team members and others. This is especially important when a new developer who is familiar with the methodology joins the project, or when he asks for help from the community.

Reviewer: suggestion (non-blocking): remove a link and slightly reduce the length
```suggestion
When working on a project based on the **Feature-Sliced Design** methodology, it is important to use original naming to benefit from the reduction of project knowledge and to make it easier to ask for help from the community.
```

Author: I rewrote this piece a little bit

## Naming Conflicts {#when-can-naming-interfere}

Naming conflicts can occur when terms used in the **Feature-Sliced Design** methodology overlap with terms used in the business:

- `FSD#process` vs simulated process in an application,
- `FSD#page` vs log page,
- `FSD#model` vs car model.

For example, a developer seeing the word process in the code will spend unnecessary time trying to understand which process they are talking about.

When communicating within the development team, when saying the word process, everyone involved should clearly understand what we're talking about, the process as a business entity or the process from Feature-Sliced Design.

When communicating with the business, developers sometimes use technical terms that the business isn't familiar with. So a developer using the term process to refer to a process from Feature-Sliced Design will introduce a misunderstanding into the conversation which may require additional time to clarify.

<!-- TODO: think of examples for other terms -->

These conflicts can disrupt the development process because developers may spend extra time trying to figure out exactly what the conversation is about.

> In addition, when communicating with the business, it is important to be aware of any technical terms that may be unfamiliar to non-technical stakeholders. Using such terms can lead to misunderstandings that may take additional time to clarify.

Reviewer: suggestion: I think we can cut down this whole section significantly:

```markdown
When the project glossary includes terms that are used in the Feature-Sliced Design methodology, extra caution should be applied in conversations with the team and non-technical stakeholders:
- When talking to the team, consider prefixing the FSD terms with the "FSD" acronym (e.g. "we can place this process on the FSD features layer")
- When talking to the business and non-technical stakeholders, consider sticking only to the project glossary and not mention the internal structure of FSD
```

I think that not only sums up everything, but also adds more information by giving specific, actionable advice

Author: I refactored this part a bit

:::note
It is important that all team members clearly understand the meaning of terms such as "process, model, ..." when communicating within the team.

> :::

Reviewer: suggestion: I'd also suggest removing this note, I think it's not so much a note as a summary of the preceding text

Author: I agree with you
